Memo rag notebook

erantr1 opened the pull request:
Added a notebook that outlines the MemoRAG technique and explains how MemoRAG works in principle.
Included an implementation of the workflow and a demonstration of the results on sample data.

erantr1 committed:
added MemoRAG

erantr1 committed:
Added Code and Usage Example to MemoRAG

erantr1 committed:
Some rearrangements

erantr1 committed:
removed memory usage in generation function

erantr1 committed:
cleaned outputs

erantr1 committed:
Added 2 more examples

erantr1 committed:
cleared outputs

NirDiamant commented:
Hi! 
very nice work. 
several things to change/add:
1) Please add documentation to the first cell of code of memory base classes
2) please write a function that displays the results and use it in each of the examples (the code focuses maximally on understanding the logic)
3) as this method uses external memory of the db and should get better results, it is crucial in this case to compare it with a vanilla RAG over the exact same queries (you can take the code from the basic tutorials in the repo) 
thanks!

erantr1 committed:
Aligned the pipeline with MemoRAG paper

erantr1 committed:
Added documentation and cleared redundant comments

erantr1 commented:
@NirDiamant 

Going through the code again I realized it may return nice answers but the pipeline/workflow was not well-aligned with the paper. So I made some changes and now I think it's much better.

I also added documentation to the renewed Memory Class and I think I now have the function you asked for in section 2 (I'm not 100% I understood what you meant there, but if I did then it's solved).

I still need to add the comparison to vanilla RAG. Will do it ASAP.

erantr1 committed:
Added comparison, not good yet

erantr1 committed:
Small fix and some commenting out

erantr1 committed:
bigger context doc

erantr1 committed:
some progress with long context

erantr1 committed:
This one works unless gpt response in memorize is not well-structured

erantr1 committed:
This one works unless gpt response in memorize is not well-structured

erantr1 committed:
1. fine-tuned prompts - made some clearer, and emphasized structure and allowed info resources
2. added pydantic and json use for memorize
3. added fallback when json is not in accurate structure.

erantr1 committed:
organized prints

erantr1 committed:
Added comparison for a short document with few questions.

NirDiamant commented:
Hi, is this ready for a review?

NirDiamant commented:
Closing due to extended inactivity. The repo has evolved significantly since this was submitted. If you'd like to contribute this, please open a fresh PR against the current main branch. Thank you for your interest in the project.
